### ISSUE TITLE
Differences to musicmetadata

### ISSUE THREAD

**Issue by kmathmann** [Borewit/music-metadata#104]
Im asking me right now what the differences between the `music-metadata` and the `musicmetadata` package are. Could you please explain the differences to me?

I tried the `musicmetadata` package and it analysed a song in 7ms for that `music-metadata` takes 28750ms. Where doese the difference from over 28 seconds came from?

With `music-metadata`
```javascript
console.time('mm');
const data = await parseStream(fileStream, mimeType, {
      duration: true,
      fileSize: size
});
console.timeEnd('mm');
console.log(data);
```
With `musicmetadata`
```javascript
console.time('mm');
mm(fileStream, {
    duration: true,
    fileSize: size
}, function (err, metadata) {
    console.timeEnd('mm');
    console.log(metadata);
});
```

**Comment by Borewit** [comment]
The differences between music-metadata and musicmetadata in a nutshell are:
- Support actual Node.js streams (instead of events)
- Supports direct file access
- It using a Promise based architecture and has been re-written in TypeScript
  - Fixes call back problems where musicmetadata is suffering from (which will take your entire application down sooner or later)
- Lots of more audio formats an audio tag headers
- Is able get much more detailed information out of the metadata
- Consistent behavior between Windows and Unix
- For every issue reported there are unit test proving correct behavior, including having an acceptable performance parsing very nasty files

But you are probably right, there is indeed a case where it does not perform very well. If you let music-metadata parse a corrupt a MPEG audio track, and you enable the duration flag to force it parsing the entire file it will cause a drastic performance degradation because it reads almost byte per byte to re-sync. The promises are becoming very heavy in this scenario. This has been captured in issue #58. I am looking for a volunteer to address this issue by the way.

For a good understanding what the differences are, maybe you should let musicmetadata parse the test-files and compare metadata of the two modules.

**Comment by kmathmann** [comment]
Thank you for that detailed answer. 

Good to hear that this issue and it's cause is known. I hope you will find a volunteer. I have currently not much time, sorry.

**Comment by Borewit** [comment]
Do you mind dropping the file which gave you that horrible performance into this issue, so I can have a look.

**Comment by kmathmann** [comment]
[05 - Jaxx & Vega vs. David Puentez vs. Mashup-Germany - We found Arcando.mp3](https://github.com/kmathmann/serve-files/raw/master/05%20-%20Jaxx%20%26%20Vega%20vs.%20David%20Puentez%20vs.%20Mashup-Germany%20-%20We%20found%20Arcando.mp3)


I'm working on windows 10 by the way.

**Comment by Borewit** [comment]
Performance issue reported as fixed in version [v2.0.3](https://github.com/Borewit/music-metadata/releases/tag/v2.0.3) in issue #103.
